pkg/endpoint/endpoint.go

```diff
@@ -461,6 +481,27 @@ type Endpoint struct {
 	// is enabled for this endpoint.
 	egressPolicyEnabled bool
 
+	// UUID is the UUID of the endpoint.
+	UUID string
+
+	// initialBuildSuccessful is true after the first build has been
```

Comment:
I'd prefer we encode data about the 'build' of the endpoint in its own structure vs. having separate fields directly within endpoint. This could include the state as well.

Reply:
I've moved it into a struct `buildStatus`. It does not change that the endpoint is currently implementing the `Buildable`. Do you have a higher level concern that we store state in the endpoint that is relative to whether the endpoint has been built?

I've separated it like this:
- Information and state regarding the currently ongoing build is in `endpointBuilder` and in the `pkg/buildqueue` package
- Information regarding the state of the endpoint continues to be in the endpoint. This includes whether the endpoint has been successfully built once, whether the last build failed, whether the endpoint is currently being built, etc. This *does not* relate to a particular build. It is the overall state considering all ongoing builds for that endpoint.

Let me know if that makes sense to you.